waterline/index.d.ts

```diff
@@ -1,33 +1,36 @@
+/// <reference types="node" />
+/// <reference types="bluebird" />
 import BluebirdPromise = require("bluebird");
 declare namespace Waterline {
-    type Adapter = Object;
-    type Connection = {
+    type Adapter = any;
+    interface Connection {
         adapter: string;
     }
     interface Config {
         adapters: { [index: string]: Adapter };
-        connections: { [index: string]: Connection }
+        connections: { [index: string]: Connection };
     }
-    type Ontology = {
+    interface Ontology {
         collections: any;
     }
     interface Waterline {
         loadCollection(collection: CollectionClass): void;
         initialize: (config: Config, cb: (err: Error, ontology: Ontology) => any) => any;
         collections: any;
     }
+    // tslint:disable-next-line:functional-interfaces
     interface CollectionClass {
-        (): Collection
+        (): Collection;
     }
     // used this comment https://github.com/balderdashy/waterline/issues/1154#issuecomment-167262575
-    export type LifecycleCallbacks = {
-        beforeValidate?: { (vaues: any, next: Function): void }[] | { (vaues: any, next: Function): void };
-        beforeCreate?: { (values: any, next: Function): void }[] | { (vaues: any, next: Function): void };
-        afterCreate?: { (newlyCreatedRecord: any, next: Function): void }[] | { (newlyCreatedRecord: any, next: Function): void };
-        beforeUpdate?: { (valuesToUpdate: any, next: Function): void }[] | { (valuesToUpdate: any, next: Function): void };
-        afterUpdate?: { (valuesToUpdate: any, next: Function): void }[] | { (valuesToUpdate: any, next: Function): void };
-        beforeDestroy?: { (criteria: any, next: Function): void }[] | { (valuesToUpdate: any, next: Function): void };
-        afterDestroy?: { (destroyedInstance: any, next: Function): void }[] | { (destroyedInstance: any, next: Function): void };
+    export interface LifecycleCallbacks {
+        beforeValidate?: Array<{ (vaues: any, next: (err?: any) => any): void }> | { (vaues: any, next: (err?: any) => any): void };
```

Comment:
Hi, @rbuckton 
Because, it is non-simple (like, string[]),, linter said: "Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead".